web/client/components/map/openlayers/Layer.jsx

```diff
@@ -123,9 +127,35 @@ class OpenlayersLayer extends React.Component {
         if (type) {
             const layerOptions = this.generateOpts(options, position, CoordinatesUtils.normalizeSRS(this.props.srs), securityToken);
             this.layer = Layers.createLayer(type, layerOptions, this.props.map, this.props.mapId);
+
             if (this.layer && !this.layer.detached) {
+                const parentMap = this.layer && this.layer.getProperties().map;
+                const mapExtent = parentMap && parentMap.getView().getProjection().getExtent();
+                const layerExtent = options && options.bbox && options.bbox.bounds;
+                const mapBboxPolygon = mapExtent && CoordinatesUtils.reprojectBbox(mapExtent, this.props.srs, 'EPSG:4326');
+                let layerBboxPolygon = layerExtent && CoordinatesUtils.getExtentFromNormalized(layerExtent, this.props.srs).extent;
+                if (layerBboxPolygon && layerBboxPolygon.length === 2 && _.isArray(layerBboxPolygon[1])) {
+                    layerBboxPolygon = layerBboxPolygon[1];
+                }
+
+                if (mapBboxPolygon && layerBboxPolygon &&
+                    !CoordinatesUtils.isBboxCompatible(CoordinatesUtils.getPolygonFromExtent(mapBboxPolygon), CoordinatesUtils.getPolygonFromExtent(layerBboxPolygon)) ||
+                    (layerOptions.type === "wmts" &&
```

Comment:
WMTS related stuff should not be implemented here, but in the WMTSLayer component

Reply:
I'm trying to check if the requested wmts layer has a projection that matches the map's current crs.
if not then it prevents the request.

Reply:
Agree, but this is the wrong place to do it, this is a generic Layer, WMTS has its own implementation in plugins/WMTS

Reply:
ok, solved